src/main/java/org/folio/service/EventService.java

```diff
@@ -0,0 +1,107 @@
+import org.folio.rest.jaxrs.model.ItemDeclaredLostEvent;
+import org.folio.rest.jaxrs.model.LoanDueDateChangedEvent;
+import org.folio.rest.persist.Criteria.Criteria;
+import org.folio.rest.persist.Criteria.Criterion;
+import org.folio.rest.persist.Criteria.Limit;
+import org.folio.rest.persist.PostgresClient;
+import org.folio.util.UuidHelper;
+
+import io.vertx.core.Future;
+
+public class EventService {
+  private static final int NUMBER_OF_EVENTS_LIMIT = 10000;
+
+  private final BaseRepository<ItemCheckedOutEvent> itemCheckedOutEventRepository;
+  private final BaseRepository<ItemCheckedInEvent> itemCheckedInEventRepository;
+  private final BaseRepository<ItemClaimedReturnedEvent> itemClaimedReturnedEventRepository;
+  private final BaseRepository<ItemDeclaredLostEvent> itemDeclaredLostEventRepository;
+  private final BaseRepository<LoanDueDateChangedEvent> loanDueDateChangedEventRepository;
+  private final BaseRepository<FeeFineBalanceChangedEvent> feeFineBalanceChangedEventRepository;
+
+  public EventService(PostgresClient postgresClient) {
+    itemCheckedOutEventRepository = new BaseRepository<>(postgresClient,
+      EventType.ITEM_CHECKED_OUT.getTableName(), ItemCheckedOutEvent.class);
+
+    itemCheckedInEventRepository = new BaseRepository<>(postgresClient,
+      EventType.ITEM_CHECKED_IN.getTableName(), ItemCheckedInEvent.class);
+
+    itemClaimedReturnedEventRepository = new BaseRepository<>(postgresClient,
+      EventType.ITEM_CLAIMED_RETURNED.getTableName(), ItemClaimedReturnedEvent.class);
+
+    itemDeclaredLostEventRepository = new BaseRepository<>(postgresClient,
+      EventType.ITEM_DECLARED_LOST.getTableName(), ItemDeclaredLostEvent.class);
+
+    loanDueDateChangedEventRepository = new BaseRepository<>(postgresClient,
+      EventType.LOAN_DUE_DATE_CHANGED.getTableName(), LoanDueDateChangedEvent.class);
+
+    feeFineBalanceChangedEventRepository = new BaseRepository<>(postgresClient,
+      EventType.FEE_FINE_BALANCE_CHANGED.getTableName(), FeeFineBalanceChangedEvent.class);
+  }
+
+  public Future<String> save(ItemCheckedOutEvent event) {
+    return itemCheckedOutEventRepository.save(event, UuidHelper.randomId());
+  }
+
+  public Future<String> save(ItemCheckedInEvent event) {
+    return itemCheckedInEventRepository.save(event, UuidHelper.randomId());
+  }
+
+  public Future<String> save(ItemClaimedReturnedEvent event) {
+    return itemClaimedReturnedEventRepository.save(event, UuidHelper.randomId());
+  }
+
+  public Future<String> save(ItemDeclaredLostEvent event) {
+    return itemDeclaredLostEventRepository.save(event, UuidHelper.randomId());
+  }
+
+  public Future<String> save(LoanDueDateChangedEvent event) {
+    return loanDueDateChangedEventRepository.save(event, UuidHelper.randomId());
+  }
+
+  public Future<String> save(FeeFineBalanceChangedEvent event) {
+    return feeFineBalanceChangedEventRepository.save(event, UuidHelper.randomId());
+  }
```

Comment:
Probably, but the repositories are not really of the same type - BaseRepository is generic, so what's the value type of the map would be? Also, Event doesn't have information about the event type, so we'll either need to add another parameter or determine it in runtime with `instanceof`.

Reply:
I was thinking of something like:
```
Map<Class<? extends Event>, BaseRepository<? extends Event>>
```
But you are right, with casting this will become overly complicated.